Rakefile

```diff
@@ -40,3 +40,15 @@ RSpec::Core::RakeTask.new(:spec) do |t|
   t.rspec_opts = '--format documentation'
 end
 
+# rake plugins
+task :plugins do
+  puts 'Registering plugins'
+  # Get name of each plugin from folder name. Require each plugin and register.
+  # At this point, we are assuming the plugin is *not* registered and is
+  # therefore not in the plugins table.
+  Dir['plugins/*'].each do |name|
```

Comment:
Okay. I did some hacking and this is a fine solution. The only way I found that I like a little more was:
```Ruby
  Dir['plugins/*/plugin.rb'].each do |name|
    require name
    # The actual name is the last part of the filename (plugin/<name>).
    Object.const_get(name.split('/')[1]).new
  end
```
But it's not difficult to understand what is happening here so I'm okay leaving it.